### PR TITLE
Update infrateam_first_responder.md

### PR DESCRIPTION
**Pull request by peetucket**
- Update instructions about testing
- Specify sul-pub specific environment

### CODE REVIEW

File: infrateam_first_responder.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ Use the `sdr-deploy` script to deploy all infrastructure projects (with **import
 
 Note that you will need to be sure you can ssh into each of the VMs from wherever you are running the deploy script.
 
-- qa: deploy to qa with script
+- qa: deploy to qa first with script, then run infrastructure-integration-tests after deploy to qa.  Note as below that some tests mail fail but will pass if run individually.  You may also need to stay on the same "space" when tests are running if using a Mac to avoid focus issues with the browser.
```
Comment (ndushay) [inline infrateam_first_responder.md:57]: "mail fail" <-- typo.   I would rephrase second sentence:  "If on a Mac, you will get better results if you stay in the same "space" as the running tests (avoids focus issues with the browser)."

**Review by ndushay — CHANGES_REQUESTED** [review]
one typo, one suggestion.

File: infrateam_first_responder.md (hunk 1 above)
Comment (peetucket) [reply infrateam_first_responder.md]: thanks, changes made.

**Comment by peetucket** [comment]
Note: I also now specified the sul-pub deploy environment that diverges from typical since it is just one and unlikely to change much now

File: infrateam_first_responder.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ Use the `sdr-deploy` script to deploy all infrastructure projects (with **import
 
 Note that you will need to be sure you can ssh into each of the VMs from wherever you are running the deploy script.
 
-- qa: deploy to qa with script
+- qa: deploy to qa first with script, then run infrastructure-integration-tests after deploy to qa.  Note as below that some tests may fail but will pass if run individually.  If on a Mac, you will get better results if you stay in the same "space" as the running tests (avoids focus issues with the browser).
```
Comment (jcoyne) [inline infrateam_first_responder.md:57]: Why run the integration tests two times in different environments? What is the objective here?  This seems like it will just be extra work for the FR, preventing them from working on other things and not deliver any additional value.

Comment (peetucket) [reply infrateam_first_responder.md:57]: so which one does it make more sense to run on...qa?  i don't have a preference, just want this to be clear from our future selves

Comment (jcoyne) [reply infrateam_first_responder.md:57]: I think that running everything on stage has been the way we have been doing it (because QA was not complete).   I don't see any reason we need to change from that practice.

Comment (peetucket) [reply infrateam_first_responder.md:57]: ok, also just noted that the H2 spec looks for public XML, which currently fails in QA since there is no purl-qa

**Comment by peetucket** [comment]
@ndushay there seems to be some disagreement about the utility of running tests in QA first, do you have a strong opinion here (i.e. suggest to revert that specific change) ... also i noted that the H2 test fails there as well due to lack of purl-qa

**Comment by ndushay** [comment]
qa is now complete enough, and running it in qa does a couple things:
1.  it ensures we notice if qa is no longer wired up correctly
2.  it is a better place to tie up with dep updates while working on any integration test changes or dep updates.  qa is closest to our old concept of "dev" servers;  stage is closest to "is this really really ready for production?"   Thus, running the tests on qa first may be a way to keep the dep updates from requiring the stage servers for a long time.

Not super strong "go to the mat" opinion;  can be overruled.

**Comment by jcoyne** [comment]
It seems reasonable to change QA to be the default place we run tests. I just don't think that we gain much by testing in both places.

**Comment by justinlittman** [comment]
I typically run tests in stage before deploy to gauge the state of the tests, deploy, and then run tests again. My preference would be to test in stage, as its initial state is likely to be more stable / less problematic for running tests than QA.

Running the tests in stage is also less disruptive to the rest of the team, as it does not require the FR to commandeer the actively used QA environment.

I definitely don't want to be running the tests in both QA and stage.

**Comment by peetucket** [comment]
Based on majority opinion, I updated instructions to indicate stage is the place to run tests first, then deploy to QA after that.

File: infrateam_first_responder.md [hunk 3]
```diff
@@ -54,12 +54,12 @@ Use the `sdr-deploy` script to deploy all infrastructure projects (with **import
...
   - It's possible that individual tests will fail due to random network connection issues, and chances are not great that the full suite will pass in one go from a spotty off-campus connection.
     - If a failed test passes upon retry against the same set of deployed codebases, that's fine.  If you're unsure whether a particular test failure indicates bad network luck, a regression in the application, or an out of date test, raise it for discussion (or another dev to retry from their laptop) in #dlss-infrastructure.
   - before deploying, warn #dlss-infra-stage-use in case there is active testing going on;  be sure to either comment out that app or coordinate with tester
-- prod: if all tests passed for stage deploys, deploy to prod with script.
+- qa: deploy to qa with script ... it is also acceptable to run infrastructure-integration-tests after deploy to qa as well or instead of in stage, it is generally not necessary to run in both qa and stage
```
Comment (ndushay) [inline infrateam_first_responder.md:61]: I would move this up with the -stage statement, but I'm not gonna block this PR over it.

**Review by ndushay — APPROVED** [review]

File: infrateam_first_responder.md (hunk 3 above)
Comment (peetucket) [reply infrateam_first_responder.md:61]: sounds good, moving it then will merge

**Review by jmartin-sul — APPROVED** [review]

**Review by jermnelson — APPROVED** [review]
